include/drivers/gpio.h

```diff
@@ -602,8 +602,8 @@ static inline int z_impl_gpio_pin_interrupt_configure(struct device *port,
 {
 	const struct gpio_driver_api *api =
 		(const struct gpio_driver_api *)port->driver_api;
-	struct gpio_driver_data *const data =
-		(struct gpio_driver_data *const)port->driver_data;
+	const struct gpio_driver_data *data =
```

Comment:
A constant pointer to mutable data is what we want. The pattern is common in the driver code, typically encoded by `DEV_DATA` macro. We could possibly do
```
const struct gpio_driver_data *const data =
```
but I don't think it's worth the trouble. We don't do it inside the drivers even if the function accesses `driver_data` read only.

Reply:
> A constant pointer to mutable data is what we want.

Well, no, it's what you want.  I think it's bizarre and unjustified, though I do see that some (not all) `DEV_DATA` macros create a const value referencing mutable data.  I'm having difficulty understanding why; the only value I can is it would stop somebody from doing:

    DEV_DATA(drv) = some_value;

I don't normally make the pointers const because people get confused and it doesn't really matter if somebody re-uses the variable with a different value, but I don't object to putting that back.  I didn't do it here because most driver adaptation headers don't do it, including other uses in this one.

I certainly do create pointers to `const` objects when I have no intention of modifying the object contents.  It lets the maintainer understand my intent.  

Are you absolutely sure you want const pointers to mutable data when the data will not be changed?  If so, I'll drop that cleanup commit and let you make the header consistent.

Reply:
Used `const` qualifier on variable declaration.